ocs_ci/ocs/resources/osd_resize.py

```diff
@@ -0,0 +1,306 @@
+import logging
+
+from ocs_ci.ocs.exceptions import (
+    StorageSizeNotReflectedException,
+    ResourceWrongStatusException,
+    CephHealthException,
+)
+from ocs_ci.ocs.resources.pod import (
+    get_osd_pods,
+    delete_pods,
+    wait_for_pods_to_be_in_statuses,
+    get_ceph_tools_pod,
+)
+from ocs_ci.ocs.resources.pvc import get_deviceset_pvcs, get_deviceset_pvs
+from ocs_ci.ocs.resources.pv import get_pv_size
+from ocs_ci.ocs.resources.storage_cluster import (
+    get_storage_size,
+    get_device_class,
+    verify_storage_device_class,
+    verify_device_class_in_osd_tree,
+)
+from ocs_ci.ocs.cluster import check_ceph_osd_tree, CephCluster
+from ocs_ci.utility.utils import ceph_health_check, TimeoutSampler, convert_device_size
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.framework import config
+
+
+logger = logging.getLogger(__name__)
+
+
+def check_resources_state_post_resize_osd(old_osd_pods, old_osd_pvcs, old_osd_pvs):
+    """
+    Check that the pods, PVCs, and PVs are in the expected state post resizing the osd.
+    It will perform the following steps:
+    1. Check that the old osd pods are in a terminating state or deleted
+    2. Check that the new osd pods running, and we have exactly the same number of osd pods as the old ones.
+    3. Check that the PVCs are in a Bound state
+    4. Check that the old PVC and PV names are equal to the current PVC and PV names
+
+    Args:
+        old_osd_pods (list): The old osd pod objects before resizing the osd
+        old_osd_pvcs (list): The old osd PVC objects before resizing the osd
+        old_osd_pvs (list): The old osd PV objects before resizing the osd
+
+    Raises:
+        ResourceWrongStatusException: If the following occurs:
+            1. The OSD pods failed to reach the status Terminated or to be deleted
+            2. The old PVC and PV names are not equal to the current PVC and PV names
+
+    """
+    old_osd_pods_count = len(old_osd_pods)
+    logger.info("Wait for the OSD pods to reach the status Terminated or to be deleted")
+    old_osd_pod_names = [p.name for p in old_osd_pods]
+    res = wait_for_pods_to_be_in_statuses(
+        expected_statuses=[constants.STATUS_TERMINATING],
+        pod_names=old_osd_pod_names,
+        timeout=300,
+        sleep=20,
+    )
+    if not res:
+        raise ResourceWrongStatusException(
+            "The OSD pods failed to reach the status Terminated or to be deleted"
+        )
+
+    logger.info("Check that the new OSD pods are running")
+    ocp_pod = OCP(kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"])
+    ocp_pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=constants.OSD_APP_LABEL,
+        resource_count=old_osd_pods_count,
+        timeout=300,
+        sleep=20,
+    )
+
+    logger.info(
+        f"Check that the number of the new OSD pods are exactly {old_osd_pods_count}"
+    )
```

Comment:
can we also add a validation that the Running OSD pod names are different than old_osd_pods names?

Reply:
Yes, I can add it.

Reply:
Done